packages/vite/src/node/plugins/resolve.ts

```diff
@@ -755,8 +755,14 @@ export function tryNodeResolve(
           }
         }
       }
+      // check if it's a self reference dep.
+      const selfPackageData = findNearestPackageData(basedir, packageCache)
+      pkg =
+        selfPackageData?.data.exports && selfPackageData?.data.name === pkgId
+          ? selfPackageData
+          : null
     }
-    return
+    if (!pkg) return
```

Comment:
I think this code should be **before** `const pkg = resolvePackageData(pkgId, basedir, preserveSymlinks, packageCache)`.

`PACKAGE_SELF_RESOLVE` (9.) is called before the resolution loop (11.).
https://nodejs.org/api/esm.html#resolution-algorithm:~:text=Let%20selfUrl%20be%20the%20result%20of%20PACKAGE_SELF_RESOLVE(packageName%2C%20packageSubpath%2C%20parentURL).
It only affects a case when you have a dep with the same name though.

Reply:
Makes sense, already updated